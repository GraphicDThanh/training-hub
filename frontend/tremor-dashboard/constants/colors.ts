type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const COLOR_CHANNELS_CHART: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};
