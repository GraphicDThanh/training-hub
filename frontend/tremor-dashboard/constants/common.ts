// Types
import { OptionType } from "@/types";

export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export enum DIRECTION {
  ASC = "asc",
  DESC = "desc",
}

// Mailto Support
export const MAILTO_SUPPORT = "support@tremor-dashboard.com";

export const REMEMBER_ME_COOKIES_KEY = "remember-me";
export const UID_KEY = "uid";
export const ROLE = "role";
export const NAME = "name";
export const AVATAR = "avatar";
export const AUTH_SESSION_COOKIES_KEY = "authjs.session-token";

export const ORDER_LIST_OPTIONS: OptionType[] = [
  { option: "Paid", value: "0" },
  { option: "Refunded", value: "2" },
  { option: "Canceled", value: "1" },
];

export const PRODUCT_LIST_OPTIONS: OptionType[] = [
  { option: "Yes", value: "true" },
  { option: "No", value: "false" },
];

export const FORMAT_DATE = {
  YEAR_MONTH_DATE_TIME: "YYYY-MM-DDTHH:mm:ss.SSS",
  MONTH_DAY_YEAR_TIME: "MMM D, YYYY h:mm A",
};

export const SEPARATOR = {
  COMMAS: ",",
  DOTS: ".",
};

export const CDN_KEY = process.env.NEXT_PUBLIC_CDN_KEY;

export const DRAG_ZONE = {
  DEFAULT: {
    TEXT: "Drag and drop your file here to begin uploading.",
    STYLE_INPUT: "border-gray-300",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_DRAG: {
    TEXT: "Release the mouse click to upload the photo",
    STYLE_INPUT: "outline outline-2 outline-[#ebeff4]",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_WRONG_FORMAT: {
    TEXT: "Invalid image file type. Please try uploading again.",
    STYLE_INPUT: "border-red-500",
    STYLE_TEXT: "text-red-500 dark:text-red-500",
  },
};

export const TRANSACTION_CLASS = {
  INCREASE: {
    buttonClasses: "text-few dark:text-few dark:hove:text-few border-few",
    textClasses: "font-semibold text-greener dark:text-few",
  },
  DECREASE: {
    buttonClasses:
      "text-attention dark:text-attention dark:hover:text-attention border-attention",
    textClasses: "font-semibold text-blood dark:text-pink",
  },
  PENDING: {
    buttonClasses:
      "text-primary dark:text-primary dark:hover:text-primary border-primary",
    textClasses:
      "font-semibold text-primary dark:text-dark-tremor-content-pending",
  },
};

export const enum VARIANT_BUTTON {
  LIGHT = "light",
  LIGHT_CARD = "lightCard",
  LIGHT_CENTER = "lightCenter",
  LIGHT_STATUS = "lightStatus",
  SECONDARY = "secondary",
  SECONDARY_SHADOW = "secondaryShadow",
  PRIMARY = "primary",
  PRIMARY_CENTER = "primaryCenter",
  SURFACE = "surface",
  DARK = "dark",
  ACTIVE = "active",
}

export const NO_IMAGE = "No Image";

export const RESULT_NOT_FOUND = "Result Not Found";

export const AMOUNT_LIMIT = 20000;
export const PRICE_LIMIT = 20000;
export const QUANTITY_LIMIT = 10000;
export const WEIGHT_LIMIT = 2000;

export const PREFIX_PHONE_NUMBER = "+1";

export const STORAGES = {
  IS_DARK_THEME: "isDarkTheme",
};

export const THEMES = {
  LIGHT: "light",
  DARK: "dark",
};

export const LOGO_TITLE = "Tremor Dashboard PRO";
