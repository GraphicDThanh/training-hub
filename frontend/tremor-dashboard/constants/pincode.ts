export const PIN_CODE_DEFAULT = "*";
export const PIN_CODE_LENGTH = 6;
export const PIN_CODE_SKIP_SET_KEY = "skipSetupPinCode";
