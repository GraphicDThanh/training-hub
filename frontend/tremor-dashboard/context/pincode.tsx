"use client";

import {
  ReactNode,
  createContext,
  useCallback,
  useContext,
  useMemo,
  useState,
} from "react";

import { updatePinCode } from "@/services";

interface IPinCodeContext {
  pinCode?: string;
  setPinCode: (value: string) => void;
}

const initialPinCodeContext: IPinCodeContext = {
  setPinCode: () => {},
};

const PinCodeContext = createContext(initialPinCodeContext);

interface IPinCodeProvider {
  children: ReactNode;
  pinCode?: string;
}

const PinCodeProvider = ({
  children,
  pinCode: initialPinCode,
}: IPinCodeProvider) => {
  const [pinCode, setPinCode] = useState<string | undefined>(initialPinCode);

  const pinCodeContextValue: IPinCodeContext = useMemo(
    () => ({ pinCode, setPinCode }),
    [pinCode],
  );

  return (
    <PinCodeContext.Provider value={pinCodeContextValue}>
      {children}
    </PinCodeContext.Provider>
  );
};

const usePinCode = () => {
  const context = useContext(PinCodeContext);

  if (!context) {
    throw new Error("usePinCode hooks should using inside PinCodeProvider!");
  }

  const { pinCode, setPinCode } = context;

  const handleSetPinCode = useCallback(
    async (pinCode: string) => {
      const { errorMessage } = await updatePinCode(pinCode);

      if (errorMessage) {
        setPinCode(pinCode);
      }

      return !!errorMessage;
    },
    [setPinCode],
  );

  return {
    pinCode,
    setPinCode: handleSetPinCode,
  };
};

export { PinCodeProvider, usePinCode };
