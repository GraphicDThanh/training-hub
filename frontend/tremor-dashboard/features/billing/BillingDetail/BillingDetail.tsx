import { memo } from "react";

// Types
import { SalaryCardData } from "@/types/billing";

// Components
import { Flex } from "@tremor/react";
import BillingCard from "../BillingCard/BillingCard";
import SalaryCard from "../SalaryCard/SalaryCard";

export interface BillingDataProps {
  cardInfo: {
    expire: string;
    cardNumber: string;
    holderFullName: string;
    cardLast4Digit: string;
  };
  aggregation: SalaryCardData[];
}

const BillingDetail = ({ cardInfo, aggregation }: BillingDataProps) => {
  const { cardNumber, holderFullName, expire, cardLast4Digit } = cardInfo;
  const billingCardNumber = cardNumber.match(/.{1,4}/g)?.join(" ") ?? "";

  return (
    <div className="w-full flex flex-col md:flex-row items-center">
      <div className="w-full mr-0 md:mr-6 max-w-[380px]">
        <BillingCard
          cardNumber={billingCardNumber}
          holderFullName={holderFullName}
          expire={expire}
          cardLast4Digit={cardLast4Digit}
        />
      </div>
      <Flex
        flexDirection="col"
        className="w-full gap-6 mt-6 md:mt-0 md:flex-row">
        {aggregation.map((item: SalaryCardData) => (
          <SalaryCard
            key={`${item.type}`}
            type={item.type}
            value={item.value}
          />
        ))}
      </Flex>
    </div>
  );
};

export default memo(BillingDetail);
