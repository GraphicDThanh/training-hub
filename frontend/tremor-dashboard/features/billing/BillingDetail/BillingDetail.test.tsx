import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";

// Components
import BillingDetail from "./BillingDetail";

// Mocks
import { AGGREGATION_TYPE, KEYBOARD_KEYS } from "@/constants";
import { PinCodeProvider } from "@/context/pincode";

const { DIGIT_1 } = KEYBOARD_KEYS;

class ResizeObserver {
  observe() {}
  unobserve() {}
  disconnect() {}
}
window.ResizeObserver = ResizeObserver;

const propsMock = (hasCardNumber: boolean) => ({
  cardInfo: {
    cardNumber: hasCardNumber ? "" : "4562 1122 4594 7852",
    holderFullName: "Jack Peterson",
    expire: "11/24",
    cardLast4Digit: "7852",
  },
  aggregation: [
    {
      value: 12324,
      type: AGGREGATION_TYPE.SALARY,
      currency: "$",
    },

    {
      value: 56789,
      type: AGGREGATION_TYPE.PAYPAL,
      currency: "$",
    },
  ],
});

describe("Testing BillingInfo component", () => {
  const setupRender = () => {
    return render(
      <PinCodeProvider pinCode="111111">
        <BillingDetail {...propsMock(false)} />,
      </PinCodeProvider>,
    );
  };

  it("Should match snapshot", () => {
    const { container } = render(<BillingDetail {...propsMock(true)} />);
    expect(container).toMatchSnapshot();
  });

  it("Should open PIN code modal when isShowAmount is false and toggleShowHide", async () => {
    const { getByTestId } = setupRender();

    fireEvent.click(getByTestId("toggle-show-amount"));

    const modal = await waitFor(() => getByTestId("modal-pincode"));
    expect(modal).toBeTruthy();
    expect(modal).toMatchSnapshot();
  });

  it("Should submit success when typing pin code correctly", async () => {
    const { getByTestId, getAllByTestId, getByText } = setupRender();

    fireEvent.click(getByTestId("toggle-show-amount"));

    await waitFor(() => getByTestId("modal-pincode"));

    const clickPinCodeField = (index: number) => {
      waitFor(() =>
        fireEvent.keyDown(getAllByTestId("pin-code-input")[index], {
          key: DIGIT_1,
        }),
      );
    };

    clickPinCodeField(0);
    clickPinCodeField(1);
    clickPinCodeField(2);
    clickPinCodeField(3);
    clickPinCodeField(4);
    clickPinCodeField(5);

    waitFor(() =>
      expect(fireEvent.click(getByText("Submit"))).toHaveBeenCalled(),
    );
  });
});
