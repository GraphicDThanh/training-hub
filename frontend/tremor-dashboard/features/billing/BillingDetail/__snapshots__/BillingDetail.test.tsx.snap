// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BillingInfo component Should match snapshot 1`] = `
<div>
  <div
    class="w-full flex flex-col md:flex-row items-center"
  >
    <div
      class="w-full mr-0 md:mr-6 max-w-[380px]"
    >
      <div
        class="tremor-Flex-root flex flex-row justify-between items-center relative text-white bg-gradient-primary dark:bg-gradient-pickled p-0 rounded-lg w-full md:w-[388px] min-h-[236px] shadow-[rgba(0,0,0,0.1)_0rem_1.25rem_1.5625rem_-0.3125rem,rgba(0,0,0,0.04)_0rem_0.625rem_0.625rem_-0.3125rem]"
      >
        <img
          alt="Master Card background"
          class="opacity-20 bg-cover object-cover"
          data-nimg="fill"
          decoding="async"
          fetchpriority="high"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fbackgrounds%2Fbg-card.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
        <div
          class="tremor-Flex-root flex w-full flex-col justify-start items-center p-4 z-10"
        >
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center"
          >
            <svg
              class="text-2xl m-2"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M1 9l2 2c4.97-4.97 13.03-4.97 18 0l2-2C16.93 2.93 7.08 2.93 1 9zm8 8l3 3 3-3a4.237 4.237 0 00-6 0zm-4-4l2 2a7.074 7.074 0 0110 0l2-2C15.14 9.14 8.87 9.14 5 13z"
              />
            </svg>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center mt-6 mb-8 gap-4"
          >
            <p
              class="text-tremor-default dark:text-dark-tremor-content text-xl text-white font-bold"
            >
              **** **** **** 7852
            </p>
            <button
              aria-label="Toggle Show Salary Button"
              class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none text-sm bg-transparent text-white hover:text-white dark:text-white dark:hover:text-white"
              data-testid="toggle-show-amount"
            >
              <svg
                class="tremor-Button-icon shrink-0 h-5 w-5"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 80c-65.2 0-118.8 29.6-159.9 67.7C89.6 183.5 63 226 49.4 256c13.6 30 40.2 72.5 78.6 108.3C169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256c-13.6-30-40.2-72.5-78.6-108.3C406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1c3.3 7.9 3.3 16.7 0 24.6c-14.9 35.7-46.2 87.7-93 131.1C433.5 443.2 368.8 480 288 480s-145.5-36.8-192.6-80.6C48.6 356 17.3 304 2.5 268.3c-3.3-7.9-3.3-16.7 0-24.6C17.3 208 48.6 156 95.4 112.6zM288 336c44.2 0 80-35.8 80-80s-35.8-80-80-80c-.7 0-1.3 0-2 0c1.3 5.1 2 10.5 2 16c0 35.3-28.7 64-64 64c-5.5 0-10.9-.7-16-2c0 .7 0 1.3 0 2c0 44.2 35.8 80 80 80zm0-208a128 128 0 1 1 0 256 128 128 0 1 1 0-256z"
                />
              </svg>
            </button>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-row justify-between items-center 2xl:py-[5px]"
          >
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center"
            >
              <div
                class="tremor-Flex-root flex w-full flex-col justify-between items-start min-w-[138px]"
              >
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-white opacity-80 tracking-[0.4px]"
                >
                  Card Holder
                </p>
                <h2
                  class="text-white font-semibold tracking-[0.12px]"
                >
                  Jack Peterson
                </h2>
              </div>
              <div
                class="tremor-Flex-root flex w-full flex-col justify-between items-start"
              >
                <p
                  class="text-tremor-default dark:text-dark-tremor-content text-white opacity-80 tracking-[0.4px]"
                >
                  Expires
                </p>
                <p
                  class="text-tremor-title dark:text-dark-tremor-content-strong text-white font-semibold tracking-[0.12px]"
                >
                  11/24
                </p>
              </div>
            </div>
            <div
              class="tremor-Flex-root flex w-full flex-row justify-end items-center"
            >
              <img
                alt="master card icon"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=96&q=75"
                srcset="/_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=48&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fmastercard.webp&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tremor-Flex-root flex flex-col justify-between items-center w-full gap-6 mt-6 md:mt-0 md:flex-row"
    >
      <div
        class="w-full font-primary antialiased items-center justify-between"
      >
        <div
          class="flex items-center"
        >
          <div
            class="tremor-Card-root w-full text-left rounded-tremor-default p-6 ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative rounded-xl shadow-md md:max-h-[236px] md:min-h-[236px]"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-center"
            >
              <div
                class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl shadow-box-icon-default w-16 h-16 bg-gradient-pickled"
              >
                <svg
                  color="white"
                  fill="currentColor"
                  height="23px"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: white;"
                  viewBox="0 0 24 24"
                  width="23px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M4 10h3v7H4zM10.5 10h3v7h-3zM2 19h20v3H2zM17 10h3v7h-3zM12 1L2 6v2h20V6z"
                  />
                </svg>
              </div>
              <div
                class="tremor-Flex-root flex w-full flex-col justify-between items-center mt-4"
              >
                <p
                  class="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold"
                >
                  Salary
                </p>
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-center items-center min-h-[35px]"
                >
                  <p
                    class="text-tremor-default text-xs text-center dark:text-dark-romance text-primary font-light"
                  >
                    Belong Interactive
                  </p>
                </div>
              </div>
            </div>
            <div
              class="tremor-Divider-root w-full mx-auto flex justify-between gap-3 items-center text-tremor-default text-tremor-content dark:text-dark-tremor-content opacity-25 dark:opacity-15 my-3"
            >
              <div
                class="w-full h-[1px] bg-tremor-border dark:bg-dark-tremor-border"
              />
            </div>
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center gap-2 min-h-[35px]"
            >
              <p
                class="text-tremor-default text-primary text-xl h-[20px] dark:text-lighter font-semibold min"
              >
                *****
              </p>
              <button
                aria-label="Toggle Show Salary Button"
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none text-sm bg-transparent text-primary hover:text-primary dark:text-lighter dark:hover:text-lighter"
                data-testid="btn-eyes"
              >
                <svg
                  class="tremor-Button-icon shrink-0 h-5 w-5"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M288 80c-65.2 0-118.8 29.6-159.9 67.7C89.6 183.5 63 226 49.4 256c13.6 30 40.2 72.5 78.6 108.3C169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256c-13.6-30-40.2-72.5-78.6-108.3C406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1c3.3 7.9 3.3 16.7 0 24.6c-14.9 35.7-46.2 87.7-93 131.1C433.5 443.2 368.8 480 288 480s-145.5-36.8-192.6-80.6C48.6 356 17.3 304 2.5 268.3c-3.3-7.9-3.3-16.7 0-24.6C17.3 208 48.6 156 95.4 112.6zM288 336c44.2 0 80-35.8 80-80s-35.8-80-80-80c-.7 0-1.3 0-2 0c1.3 5.1 2 10.5 2 16c0 35.3-28.7 64-64 64c-5.5 0-10.9-.7-16-2c0 .7 0 1.3 0 2c0 44.2 35.8 80 80 80zm0-208a128 128 0 1 1 0 256 128 128 0 1 1 0-256z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full font-primary antialiased items-center justify-between"
      >
        <div
          class="flex items-center"
        >
          <div
            class="tremor-Card-root w-full text-left rounded-tremor-default p-6 ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand bg-tremor-primary dark:bg-dark-tremor-primary mx-auto pt-3 pb-4 px-4 ring-0 max-w-full lg:max-w-[356px] 2xl:max-w-full border-none relative rounded-xl shadow-md md:max-h-[236px] md:min-h-[236px]"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-center"
            >
              <div
                class="tremor-Flex-root flex flex-row items-center p-1 justify-center rounded-xl shadow-box-icon-default w-16 h-16 bg-gradient-pickled"
              >
                <svg
                  color="white"
                  fill="currentColor"
                  height="18px"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: white;"
                  viewBox="0 0 24 24"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21.4945038,7.054125 C20.4250038,12.026625 17.0155038,14.658375 11.6050038,14.658375 L9.64300376,14.658375 L8.17975376,24.000375 L11.3620038,24.000375 C11.8217538,24.000375 12.2125038,23.665875 12.2845038,23.212125 L12.3220038,23.014125 L13.0540038,18.377625 L13.1012538,18.121875 C13.1725038,17.668125 13.5640038,17.333625 14.0230038,17.333625 L14.6042538,17.333625 C18.3640038,17.333625 21.3085038,15.805875 22.1687538,11.388375 C22.5137538,9.615375 22.3480038,8.128875 21.4945038,7.054125 M19.3172538,1.80975 C18.2057538,0.543 16.1972538,0 13.6270038,0 L6.16900376,0 C5.64325376,0 5.19625376,0.3825 5.11375376,0.90075 L2.00800376,20.59725 C1.94650376,20.98575 2.24725376,21.3375 2.64100376,21.3375 L7.24525376,21.3375 L8.40175376,14.0025 L8.36575376,14.232 C8.44825376,13.71375 8.89225376,13.33125 9.41725376,13.33125 L11.6050038,13.33125 C15.9040038,13.33125 19.2692538,11.58525 20.2525038,6.53475 C20.2817538,6.38475 20.3290038,6.0975 20.3290038,6.0975 C20.6080038,4.2285 20.3267538,2.96025 19.3172538,1.80975"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <div
                class="tremor-Flex-root flex w-full flex-col justify-between items-center mt-4"
              >
                <p
                  class="mb-1 text-primary dark:text-lighter text-tremor-title leading-[33px] tracking-[0.1764px] font-semibold"
                >
                  Paypal
                </p>
                <div
                  class="tremor-Flex-root flex w-full flex-row justify-center items-center min-h-[35px]"
                >
                  <p
                    class="text-tremor-default text-xs text-center dark:text-dark-romance text-primary font-light"
                  >
                    Freelance Payment
                  </p>
                </div>
              </div>
            </div>
            <div
              class="tremor-Divider-root w-full mx-auto flex justify-between gap-3 items-center text-tremor-default text-tremor-content dark:text-dark-tremor-content opacity-25 dark:opacity-15 my-3"
            >
              <div
                class="w-full h-[1px] bg-tremor-border dark:bg-dark-tremor-border"
              />
            </div>
            <div
              class="tremor-Flex-root flex w-full flex-row justify-between items-center gap-2 min-h-[35px]"
            >
              <p
                class="text-tremor-default text-primary text-xl h-[20px] dark:text-lighter font-semibold min"
              >
                *****
              </p>
              <button
                aria-label="Toggle Show Salary Button"
                class="tremor-Button-root shrink-0 inline-flex justify-center items-center group font-medium outline-none text-sm bg-transparent text-primary hover:text-primary dark:text-lighter dark:hover:text-lighter"
                data-testid="btn-eyes"
              >
                <svg
                  class="tremor-Button-icon shrink-0 h-5 w-5"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M288 80c-65.2 0-118.8 29.6-159.9 67.7C89.6 183.5 63 226 49.4 256c13.6 30 40.2 72.5 78.6 108.3C169.2 402.4 222.8 432 288 432s118.8-29.6 159.9-67.7C486.4 328.5 513 286 526.6 256c-13.6-30-40.2-72.5-78.6-108.3C406.8 109.6 353.2 80 288 80zM95.4 112.6C142.5 68.8 207.2 32 288 32s145.5 36.8 192.6 80.6c46.8 43.5 78.1 95.4 93 131.1c3.3 7.9 3.3 16.7 0 24.6c-14.9 35.7-46.2 87.7-93 131.1C433.5 443.2 368.8 480 288 480s-145.5-36.8-192.6-80.6C48.6 356 17.3 304 2.5 268.3c-3.3-7.9-3.3-16.7 0-24.6C17.3 208 48.6 156 95.4 112.6zM288 336c44.2 0 80-35.8 80-80s-35.8-80-80-80c-.7 0-1.3 0-2 0c1.3 5.1 2 10.5 2 16c0 35.3-28.7 64-64 64c-5.5 0-10.9-.7-16-2c0 .7 0 1.3 0 2c0 44.2 35.8 80 80 80zm0-208a128 128 0 1 1 0 256 128 128 0 1 1 0-256z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing BillingInfo component Should open PIN code modal when isShowAmount is false and toggleShowHide 1`] = `
<div
  aria-modal="true"
  class="tremor-dialog-root relative z-50"
  data-headlessui-state="open"
  data-testid="modal-pincode"
  id="pincode-modal"
  role="dialog"
>
  <div
    class="fixed inset-0 bg-slate-950/30 dark:bg-slate-950/50 transition-opacity ease-out duration-300 opacity-0"
  />
  <div
    class="fixed inset-0 overflow-y-auto w-screen"
  >
    <div
      class="flex min-h-full items-center justify-center p-4"
    >
      <div
        class="tremor-dialog-panel max-w-lg overflow-hidden text-left ring-1 shadow-tremor transition-all transform text-tremor-content ring-tremor-ring dark:text-dark-tremor-content dark:ring-dark-tremor-ring rounded-tremor-default p-6 rounded-md border-none outline-none flex flex-col gap-8 sm:min-w-[80%] bg-tremor-primary dark:bg-dark-tremor-primary md:min-w-[390px] w-[calc(100%-8px)] md:max-w-[390px] ease-out duration-300 opacity-0 scale-95"
        data-headlessui-state="open"
        id="headlessui-dialog-panel-:rd:"
      >
        <h2
          class="text-primary dark:text-white font-bold text-2xl text-center"
        >
          Enter your PIN code
        </h2>
        <div>
          <div
            class="tremor-Flex-root flex flex-row justify-between items-center gap-3 sm:gap-6 w-full self-center max-w-[250px] sm:max-w-[360px] m-auto"
          >
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent focused"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
            <input
              class="w-8 sm:w-full h-8 sm:h-10 border-2 rounded-md outline-none border-secondary focus:border-tremor-secondary text-center font-bold text-2xl text-primary dark:text-white placeholder-secondary focus:placeholder-transparent dark:bg-transparent dark:border-white focus:dark:border-tremor-secondary dark:placeholder-white focus:dark:placeholder-transparent"
              data-testid="pin-code-input"
              inputmode="numeric"
              maxlength="1"
              placeholder="○"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-evenly items-center gap-5"
        >
          <button
            class="tremor-Button-root inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted border-tremor-brand dark:border-dark-tremor-brand hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis opacity-50 cursor-not-allowed py-3 px-6 rounded-lg border-none bg-gradient-primary dark:bg-gradient-pickled shadow-btn-primary dark:!shadow-btn-primary hover:!shadow-btn-primary-hover dark:hover:!shadow-btn-primary-hover flex-1 min-w-[64px] sm:px-[22px]"
            disabled=""
          >
            <span
              class="tremor-Button-text text-tremor-default whitespace-nowrap"
            >
              <p
                class="text-tremor-default flex items-center uppercase py-[2px] text-xs font-bold text-white dark:text-dark-tremor-content-title tracking-wide"
              >
                Submit
              </p>
            </span>
          </button>
          <button
            class="tremor-Button-root inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default border shadow-tremor-input dark:shadow-dark-tremor-input text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis py-3 px-6 rounded-lg border-none bg-gradient-btn-back dark:bg-gradient-btn-back shadow-btn-surface dark:!shadow-btn-surface hover:!shadow-btn-surface-hover dark:hover:!shadow-btn-surface-hover mt-0 flex-1"
            data-testid="btn-close-modal"
            type="submit"
          >
            <span
              class="tremor-Button-text text-tremor-default whitespace-nowrap"
            >
              <p
                class="text-tremor-default uppercase font-bold text-xs text-gray-900 dark:text-black tracking-wide"
              >
                Cancel
              </p>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
